GNUmakefile

```diff
@@ -0,0 +1,50 @@
+# Makefile for the supdup server and client.
+
+SHELL = /bin/sh
+PREFIX ?= /usr/local
+
+OS_NAME = $(shell uname)
+
+CC ?= cc
+CFLAGS += $(shell pkg-config --cflags ncurses 2> /dev/null) -g -Wall
+LDFLAGS += -g
+
+# AIX needs linking libcurses *after* libncurses
+ifeq ($(OS_NAME), AIX)
+ EXTRA_LDFLAGS = -lcurses
+endif
+
+# Mac OSX
+ifeq ($(OS_NAME), Darwin)
+ LDFLAGS = -L/opt/local/lib
+endif
+
+# Haiku
+ifeq ($(OS_NAME), Haiku)
+ EXTRA_LDFLAGS = -lnetwork
+endif
+
+# NetBSD
+ifeq ($(OS_NAME), NetBSD)
+ CFLAGS += -I/usr/pkg/include -I/usr/pkg/include/ncurses
+ LDFLAGS += -L/usr/pkg/lib
+endif
+
+# The server isn't ready for prime time.
+all:	supdup
+
+SUPDUP_OBJS = supdup.o charmap.o tcp.o chaos.o
+supdup: $(SUPDUP_OBJS)
+	$(CC) $(LDFLAGS) -o $@ $(SUPDUP_OBJS) $(shell pkg-config --libs ncurses 2> /dev/null || printf '%s' "-lncurses") $(EXTRA_LDFLAGS)
```

Comment:
Yuck .. but maybe put this `$(shell peg-config ...)` stuff into a variable, and then use it... With a small comment on why this ugly stuff. :-)

Reply:
Will do